tests/Generator/Caller/SimpleCallerTest.php

```diff
@@ -0,0 +1,189 @@
+use Nelmio\Alice\Throwable\GenerationThrowable;
+use Prophecy\Argument;
+
+/**
+ * @covers \Nelmio\Alice\Generator\Caller\SimpleCaller
+ */
+class SimpleCallerTest extends \PHPUnit_Framework_TestCase
+{
+    public function testIsACaller()
+    {
+        $this->assertTrue(is_a(SimpleCaller::class, CallerInterface::class, true));
+    }
+
+    public function testIsValueResolverAware()
+    {
+        $this->assertEquals(
+            (new SimpleCaller())->withValueResolver(new FakeValueResolver()),
+            new SimpleCaller(new FakeValueResolver())
+        );
+    }
+
+    /**
+     * @expectedException \Nelmio\Alice\Throwable\Exception\Generator\Resolver\ResolverNotFoundException
+     * @expectedExceptionMessage Expected method "Nelmio\Alice\Generator\Caller\SimpleCaller::doCallsOn" to be called only if it has a resolver.
+     */
+    public function testThrowsAnExceptionIfDoesNotHaveAResolver()
+    {
+        $obj = new FakeObject();
+
+        $caller = new SimpleCaller();
+        $caller->doCallsOn($obj, ResolvedFixtureSetFactory::create(), new GenerationContext());
+    }
+
+    public function testCallsMethodsOntoTheGivenObject()
+    {
+        $dummyProphecy = $this->prophesize(Dummy::class);
+        $dummyProphecy->setTitle('foo_title')->shouldBeCalled();
+        $dummyProphecy->addFoo()->shouldBeCalledTimes(2);
+        $dummy = $dummyProphecy->reveal();
+
+        $object = new SimpleObject('dummy', $dummy);
+        $set = ResolvedFixtureSetFactory::create(
+            null,
+            $fixtures = (new FixtureBag())->with(
+                new SimpleFixture(
+                    'dummy',
+                    Dummy::class,
+                    new SpecificationBag(
+                        null,
+                        new PropertyBag(),
+                        (new MethodCallBag())
+                            ->with(new SimpleMethodCall('setTitle', [ 'foo_title' ]))
+                            ->with(new SimpleMethodCall('addFoo'))
+                            ->with(new SimpleMethodCall('addFoo'))
+                    )
+                )
+            )
+        );
+        $context = new GenerationContext();
+        $context->markIsResolvingFixture('foo');
+
+        $caller = new SimpleCaller(new FakeValueResolver());
+        $caller->doCallsOn($object, $set, $context);
+    }
+
+    public function testResolvesAllPropertyValues()
+    {
+        $object = new SimpleObject('dummy', new Dummy());
+        $set = ResolvedFixtureSetFactory::create(
+            null,
+            $fixtures = (new FixtureBag())->with(
+                $fixture = new SimpleFixture(
+                    'dummy',
+                    Dummy::class,
+                    new SpecificationBag(
+                        null,
+                        new PropertyBag(),
+                        (new MethodCallBag())
+                            ->with(new SimpleMethodCall('setTitle', [$titleValue = new FakeValue()]))
```

Comment:
I would add a second call on `setTitle` to make sure the second overrides the first